internal/log/context.go

```diff
@@ -25,22 +25,60 @@ func ContextWithAttrs(ctx context.Context, attr ...slog.Attr) context.Context {
 	return context.WithValue(ctx, contextAttrSliceKey, attrSlice)
```

Comment:
Also not in this PR but suggest renaming `log.ContextWithAttrs(ctx)` to `log.AppendAttrs(ctx)` to piggyback off the semantics of `append()`. 

Also nit: replace lines 23 and 24 with `attrSlice := append(attrSliceFromContext(ctx), attr...)`

Reply:
Okay, I just think `ContextWithAttrs` is a little verbose but it's not a strong opinion. Happy to leave it.